src/OrchardCore.Modules/OrchardCore.Rules/Drivers/JavascriptConditionDisplayDriver.cs

```diff
@@ -40,8 +61,39 @@ public override async Task<IDisplayResult> UpdateAsync(JavascriptCondition condi
             var model = new JavascriptConditionViewModel();
             if (await updater.TryUpdateModelAsync(model, Prefix))
             {
-                // TODO is empty.
-                condition.Script = model.Script;
+                if (string.IsNullOrWhiteSpace(model.Script))
+                {
+                    updater.ModelState.AddModelError(Prefix, nameof(model.Script), S["The script is required"]);
+                    //Codemirror hides the textarea which displays the error when updater.ModelState.AddModelError is used, that's why a notifier is used to show the error to user
+                    await _notifier.ErrorAsync(H["The script is required"]);
+                    return Edit(condition);
+                }
+
+                try
+                {
+                    _ = await _evaluator.EvaluateAsync(new()
```

Comment:
use ParseScript instead of evaluate

Reply:
Should I parse it first and then evaluate? Sébastien Roses second point in the issue is "ensure that evaluating the script is not breaking the full page..." so I have to evaluate and I thought that since EvaluateAsync parses it with ParseScript before evaluation then I only need to catch the exception, otherwise I would be parsing it twice.